src/api/sourceApi.ts

```diff
@@ -93,11 +93,22 @@ const getById = (id, t, params = {}) => {
         });
 };
 
+    const checkSource = (source) => {
+        return axios.get('/check-source', {params: {source}})
+        .then((response) => {
+            return response.data;
+        })
```

Comment:
suggestion: we should verify the response's status code, and throw an error if it's equal to 404 or 500